### ISSUE TITLE
Yup resolver failing to validate one field dependent on another

### ISSUE THREAD

**Issue by Mathius17** [react-hook-form/resolvers#27]
Hi there! First of all I'm not sure if this problem is caused by the resolver, if it's a problem with Yup itself or simply because I don't have a proper understanding of the usage of this two wonderful tools.

So I'm trying to validate a field based on the value of another field (comparing that last date is greater than first date). But the error messages have been rather inconsistent in this case, usually not disappearing when it should. I made a [sandbox](https://codesandbox.io/s/polished-framework-ofvx8) to reproduce the issue in a simplified way (two numbers, first one should be less than the second one).

Working sandbox: [https://codesandbox.io/s/polished-framework-ofvx8](https://codesandbox.io/s/polished-framework-ofvx8)

**Steps to reproduce:**
1. Insert a number in the first input
2. Insert a lower number in the second input (you should get an error message)
3. Change the first input so that first < second
4. Blur

**Desired outcome:**
After step 3, the error message on the second input should disappear.

**Current outcome:**
The error message on the second input remains even though it went through the validation (as showcased in the console). Also note that even if the error message doesn't disappear, the "submit" is now clickable as the form is valid (disabled is form invalid).

**Comment by Vandell63** [comment]
Hey @Mathius17  i have check your code and i found some issue and misunderstood of how yup work when you want to check two values.
1. you should remove your schema from your component
2. you should test your logic in the schema directly

I suggest you an alternative to your code their -> https://codesandbox.io/s/infallible-panini-uz2ri?file=/src/App.js
Hope your understand the difference and the yup logic behind.

**Comment by Mathius17** [comment]
Thank you very much @Vandell63! I definitely like your approach way more and now I have a better understanding of how to check two fields dependent on each other in Yup.

One last question just to make sure we're on the same page. The issue still remains, right? Although is a problem with Yup itself.

**Comment by Vandell63** [comment]
No sorry, your issue is fixed for me.
i just got another another issue with yup about context object inject into the form. Sorry it is independent from your case.
you case i think we have resolve it right ?

**Comment by Mathius17** [comment]
@Vandell63 I just checked your alternative code again and repeating the "Steps to reproduce" in your example is giving me the same results unfortunately.

**Comment by Vandell63** [comment]
Yes i see now sorry. You're right !. The error disappear only when you change the `Max` value. But when you change the min value only the `isValid` property trigger. But the error don't disappear. 
So point 1 and 2 correct but not the third one. :/ . I continue to investigate

**Comment by bluebill1049** [comment]
Thanks, guys, will investigate the cause of the issue.

**Comment by bluebill1049** [comment]
react hook form validate a single field at a time (this allow us to optimize render), if you need to depend field to trigger validation, you can use `trigger`.

**Comment by Mathius17** [comment]
Thanks @bluebill1049! I'm closing the issue then.
